### ISSUE TITLE
Support yaml as a config file

### ISSUE THREAD

**Issue by umi0410** [yorkie-team/yorkie#223]
**What would you like to be added**:
I'd like to yorkie support `.yaml` or other config formats. 
**Why is this needed**:

https://github.com/yorkie-team/yorkie/blob/7c254bb96d68b1f4db4164901918f2acaf8d5867/yorkie/config.go#L81

`yorkie` supports only `json` config files. However, `yaml` is also a good format to contains application configurations so it would be nice if `yaml` is supported.

**Comment by hackerwins** [comment]
Thanks for the good comments. It seems to use YAML more than JSON for the config file.
But I don't think it's a good idea to support two formats(JSON, YAML) for config.

I have two opinions.
1. How about replacing JSON config with YAML config?
2. We have [a config file example](https://github.com/yorkie-team/yorkie/blob/main/yorkie/config.sample.json), but it lacks entries and descriptions. How about adding all the fields and descriptions like [etcd](https://github.com/etcd-io/etcd/blob/main/etcd.conf.yml.sample)?

**Comment by umi0410** [comment]
I agree with you and basically prefer `yaml` :smile: . It may also be troublesome to manage multiple sample config files in different formats.
And your second opinion looks great because `yaml` support comments,

**Comment by hackerwins** [comment]
After that, we can add descriptions to [the sample YAML file](https://github.com/yorkie-team/yorkie/blob/main/yorkie/config.sample.yml) then close the issue.

**Comment by parkjh688** [comment]
I have no idea what kind of comments to write, need help!

`RPC` : For Distributed System
`Metrics`: 
`Mongo`: For content differences between documents and snapshots  
`ETCD`:
`Backend`:
